File: SnakesAndLadders/Player.cs
public class Player
{

    public Player(string name)
    {
        Name = name;
    }
    public string Name { get; set; }
    public int CurrentPosition { get; set; } = 0;
    public bool IsWinner => CurrentPosition == 100;

    public int MoveToNewPosition(int increment)
    {
        if(CurrentPosition + increment > 100)
        {
            return CurrentPosition;
        }

        return CurrentPosition + increment;
    }

    public void SetPosition(int position)
    {
        CurrentPosition = position;
    }

    public int GetPosition()
    {
        return CurrentPosition;
    }


}

File: src/main/java/org/hsharan/components/Player.java
package org.hsharan.components;

import java.util.*;

public class Player {
    private String name;
    private Integer id;

    private Integer position;

    public Integer getPosition() {
        return position;
    }

    public void setPosition(Integer position) {
        this.position = position;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    private static Map<Integer,Player> players = new LinkedHashMap<>();


    private static Integer idCounter=0;

    public Player(String name) {
        this.name = name;
        idCounter+=1;
        this.id= idCounter;

        players.put(this.id,this);
    }



    public static List<Player> getAllPlayer() {
        return new ArrayList<Player>(players.values());
    }
}
